includes/Context.php

```diff
@@ -357,14 +357,32 @@ public function get_amp_mode() {
 			return false;
 		}
 
-		$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
-			&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
+		$amp_plugin_version       = AMP__VERSION;
+		$amp_exposes_support_mode = defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
 			&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
 			&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
 
+		if ( strpos( $amp_plugin_version, '-' ) !== false ) {
+			$amp_plugin_version = explode( '-', $amp_plugin_version )[0];
+		}
+
+		$amp_plugin_version_2_or_higher = version_compare( $amp_plugin_version, '2.0.0', '>=' );
+
+		if ( $amp_plugin_version_2_or_higher ) {
+			$exposes_support_mode = method_exists( 'AMP_Options_Manager', 'get_option' )
+				&& $amp_exposes_support_mode;
+		} else {
+			$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
+				&& $amp_exposes_support_mode;
+		}
+
 		if ( $exposes_support_mode ) {
 			// If recent version, we can properly detect the mode.
-			$mode = AMP_Theme_Support::get_support_mode();
+			if ( $amp_plugin_version_2_or_higher ) {
+				$mode = \AMP_Options_Manager::get_option( 'theme_support' );
```

Comment:
Nit-picking, but we can remove the backslash from here and instead add a `use` statement for `AMP_Options_Manager` on top of the file.

Reply:
Ah, sure thing 😄